cluster/cluster.go

```diff
@@ -79,20 +79,31 @@ func Start() {
 	log.Info("CLU Start: joined to %d nodes in cluster", n)
 }
 
+type partitionCandidates struct {
+	priority int
+	nodes    []Node
+}
+
 // return the list of nodes to broadcast requests to
-// Only 1 member per partition is returned. This list includes
-// ThisNode if it is capable of handling queries.
+// Only 1 member per partition is returned.
+// The nodes are selected based on priority, using a random selection
+// if there are multi nodes with the highest priority.
 func MembersForQuery() []Node {
 	thisNode := Manager.ThisNode()
 	// If we are running in single mode, just return thisNode
 	if Mode == ModeSingle {
 		return []Node{thisNode}
 	}
 
-	membersMap := make(map[int32][]Node)
+	// store the available nodes for each partition, grouped by
+	// priority
+	membersMap := make(map[int32]*partitionCandidates)
 	if thisNode.IsReady() {
```

Comment:
No.  Nodes that are marked as notReady are still excluded.  The priority is used in addition to the ready state (set to ready after warmUpPeriod).  It is implemented this way to ensure that when using carbon input, things work exactly as they previously did as all nodes will have the same prio.

having a small warmUp period is still useful as the nodes do a fair amount of work during startup. 

This also keeps our options open for later marking the node as notReady based on other information (query failure rates,  manually via the api, etc..)